### PR TITLE
[release-2.9] Update integration tests for Placement

### PR DESCRIPTION
**Pull request by dhaiducek**
Selective cherry-pick of:
- #681
- #682
- #683

Specifically, this PR:
- Removes allowList from appsub
- Adds ManagedClusterSetBindings to generator appsub

ref:
- https://github.com/stolostron/backlog/issues/27443
- https://github.com/stolostron/backlog/issues/27444

### CODE REVIEW

**Comment by dhaiducek** [comment]
Blocked until the hosted mode cherry-pick is merged:
- #694

**Review by JustinKuli — APPROVED** [review]

**Pull request merged**
